### ISSUE TITLE
Distribute Strategy wrong reduction

### ISSUE THREAD

**Issue by EmanueleGhelfi** [tensorflow/tensorflow#28956]
## Description of issue (what needs changing):
In the documentation the snippet:

```python
def train_step():
  def step_fn(inputs):
    features, labels = inputs
    logits = model(features)
    cross_entropy = tf.nn.softmax_cross_entropy_with_logits(
        logits=logits, labels=labels)  
    loss = tf.reduce_sum(cross_entropy) * (1.0 / global_batch_size)
    train_op = optimizer.minimize(loss)
    with tf.control_dependencies([train_op]):
      return tf.identity(loss)

  per_replica_losses = mirrored_strategy.experimental_run(
      step_fn, input_iterator)
  mean_loss = mirrored_strategy.reduce(
      tf.distribute.ReduceOp.MEAN, per_replica_losses)
  return mean_loss
```
calculates the loss using the distribution strategy. However the loss for each replica is reduced using `tf.distribute.ReduceOp.MEAN`. I think that the correct loss to return is the loss reduced using `tf.distribute.ReduceOp.SUM` since every loss is already reduced using the partial mean over the `global_batch_size` (` tf.reduce_sum(cross_entropy) * (1.0 / global_batch_size)`) . 

I think a better snippet would be:

```python
def train_step():
  def step_fn(inputs):
    features, labels = inputs
    logits = model(features)
    cross_entropy = tf.nn.softmax_cross_entropy_with_logits(
        logits=logits, labels=labels)  
    loss = tf.reduce_sum(cross_entropy) * (1.0 / global_batch_size)
    train_op = optimizer.minimize(loss)
    with tf.control_dependencies([train_op]):
      return tf.identity(loss)

  per_replica_losses = mirrored_strategy.experimental_run(
      step_fn, input_iterator)
  loss = mirrored_strategy.reduce(
      tf.distribute.ReduceOp.SUM, per_replica_losses)
  return loss
```

Am I wrong?

### Submit a pull request?

Maybe I can submit a pull request to fix this.

Are you planning to also submit a pull request to fix the issue? See the docs
contributor guide: https://www.tensorflow.org/community/contribute/docs and the
docs style guide: https://www.tensorflow.org/community/contribute/docs_style

**Comment by guptapriya** [comment]
hi @EmanueleGhelfi - you're absolutely right, and looks like we've fixed the documentation too :) thanks for reporting and apologies for the delay!
